src/main/java/edu/ucdavis/dss/ipa/services/jpa/JpaInstructorTypeCostService.java

```diff
@@ -0,0 +1,66 @@
+package edu.ucdavis.dss.ipa.services.jpa;
+
+import edu.ucdavis.dss.ipa.entities.InstructorTypeCost;
+import edu.ucdavis.dss.ipa.repositories.InstructorTypeCostRepository;
+import edu.ucdavis.dss.ipa.services.InstructorCostService;
+import edu.ucdavis.dss.ipa.services.InstructorTypeCostService;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import javax.inject.Inject;
+import java.util.List;
+
+@Service
+public class JpaInstructorTypeCostService implements InstructorTypeCostService {
+    @Inject
+    InstructorTypeCostRepository instructorTypeCostRepository;
+    @Inject InstructorCostService instructorCostService;
+
+    @Override
+    public List<InstructorTypeCost> findByBudgetId(Long budgetId) {
+        return instructorTypeCostRepository.findByBudgetId(budgetId);
+    }
+
+    @Override
+    public InstructorTypeCost findById(Long instructorTypeId) {
+        return instructorTypeCostRepository.findById(instructorTypeId);
+    }
+
+    @Override
+    @Transactional
+    public void deleteById(long instructorTypeId) {
+        InstructorTypeCost instructorTypeCost = this.findById(instructorTypeId);
+
+        instructorCostService.removeAssociationByInstructorTypeId(instructorTypeId);
+
+        instructorTypeCostRepository.deleteById(instructorTypeId);
+    }
+
+    @Override
+    public InstructorTypeCost update(InstructorTypeCost newInstructorType) {
```

Comment:
Why does this function take an InstructorTypeCost, then fetch it again, then update it?

Reply:
This does two things:
- Defensive coding to ensure the object does exist before attempting to modify it.
- Ensuring that only explicit fields are changed, regardless of what the DTO (user) may have done with the payload. In this case, budgetId and instructorId should never change, only cost can be modified.

Reply:
Could you document that update() does not update the object but merely ensures it exists and copies the cost?

This is a code smell imo (given the function is merely called 'update') but I'll defer to you.